synapse/lib/remotelayer.py

```diff
@@ -0,0 +1,59 @@
+'''
+The layer that exists locally (as a client) passing data requests to a remote layer
+'''
+import logging
+
+import synapse.exc as s_exc
+import synapse.telepath as s_telepath
+
+import synapse.lib.layer as s_layer
+
+logger = logging.getLogger(__name__)
+
+PASSTHROUGHFUNCS = (
+    'commit', 'getBuidProps', 'getOffset', 'setOffset', 'stat', 'initdb', 'stor', 'splicelistAppend',
+)
+
+class RemoteLayer(s_layer.Layer):
+    '''
+    A layer retrieved over telepath.
+    '''
+    confdefs = (  # type: ignore
+        ('remote:telepath', {'type': 'str', 'doc': 'Path to remote layer'}),
+    )
+
+    async def __anit__(self, dirn, readonly=False):
+
+        await s_layer.Layer.__anit__(self, dirn, readonly=readonly)
+
+        self.path = self.conf.get('remote:telepath')
+        if self.path is None:
+            raise s_exc.BadConfValu('Missing remote layer path')
+
+        self.remote = await s_telepath.openurl(self.path)
+        self.onfini(self.remote.fini)
```

Comment:
If there is a network blip, our proxy will be fini'd.  This will prevent using the cortex process which is otherwise online.
Do we want to add a onfini handler to self.proxy which will attempt to reconnect every N seconds or so?  It feels like having some level of self healing is preferable over having to potentially kick a bunch of containers.

Reply:
That's a point for discussion.

Reply:
After further discussion, we're going to punt this to a separate story to handle this problem in a more general way.